### DISCUSSION TITLE
Grafana Dashboard

### DISCUSSION THREAD

**Discussion by ezigold** [johanmeijer/grott#69] [Show and tell]
Awesome job Johan THANK YOU :)
I have finally found some code for growatt inverter data that actually works :)

I have used it and can now visualise my inverter data using grafana on my RPi4

![grott=grafana-dashboard-growatt-spf5000es](https://user-images.githubusercontent.com/83513270/135268826-f7b4e8f2-b849-44f8-8530-777c66a39bd9.jpg)

**Comment by johanmeijer** [comment]
Thanks!

Look nice. I lost my grafana dashboard due to RPI chrash. Advise: don't forget to make a backup! RPI's (sd cards) are very brittle.

**Comment by Dandiiik** [comment]
Very very nice, should this setting be obtained?

**Comment by the-butterfry** [comment]
Here I'll share my Grafana dash too. Happy to share the code - or something you can adapt to your data. 
![image](https://user-images.githubusercontent.com/37050797/160177249-8aaae076-5cef-4bce-99e6-3ab0e4478127.png)

**Reply by Dandiiik** [reply]
Yours is very nice too. I can not do it. Can I ask you to share your settings?

**Reply by the-butterfry** [reply]
Why don't you start with @[johanmeijer](https://github.com/johanmeijer) dashboard below. It will give you the items you need that grott supplies. My dashboard mostly includes data from my Bluetooth BMS (jbd / Overkill solar bms) and wouldn't be useful unless you were using that system too.

**Comment by johanmeijer** [comment]
I created a simple example dashboard (absolutely not that fancy as the ones above but nice for a start). 

![image](https://user-images.githubusercontent.com/16508298/160281481-17b55253-daf8-4e41-b1f7-edc4e31315fb.png)

You can import this file to start with. You can find it here:

https://github.com/johanmeijer/grott/tree/master/examples/Grafana

My own dashboard looks a little bit different but there is also information of Energy Usage, Gas Usaga and my 20+ year old solar panels on it. 

![image](https://user-images.githubusercontent.com/16508298/160281760-a519a503-1a7a-4b49-81b3-b20666a41fdc.png)

**Reply by Dandiiik** [reply]
Thank you, your example helped me a lot in setting up my charts. But the outputs do not work quite well. I have 3x SPF5000E
![2022-04-03](https://user-images.githubusercontent.com/62666350/161446977-e1698d10-5f78-403a-8449-9d634a59b279.png)

**Comment by johanmeijer** [comment]
Ok interesting. You have 3x (3 different) SPF inverters? Do  they have al separate datalogers (shinewifi/shinelan) or do hava a shinebox with RF dataloggers? 

I do not a have a lot of experience with multiple inverters. In influx/grafana this should give 3 measurements with the different values? Do you see this? 

The fields for a SPF  are different. This should be the values you can select. 
    
            "decrypt"           
            "datalogserial"  
            "pvserial"          
            "pvstatus"          
            "vpv1"              
            "vpv2"              
            "ppv1"             
            "ppv2"             
            "buck1curr"     
            "buck2curr"     
            "op_watt"        
            "pvpowerout"   
            "op_va"             
            "acchr_watt"     
            "acchr_VA"       
            "bat_Volt"        
            "batterySoc"    
            "bus_volt"        
            "grid_volt"       
            "line_freq"       
            "outputvolt"    
            "pvgridvoltage"   
            "outputfreq"        
            "invtemp"           
            "dcdctemp"        
            "loadpercent"     
            "buck1_ntc"        
            "buck2_ntc"        
            "OP_Curr"           
            "Inv_Curr"          
            "AC_InWatt"      
            "AC_InVA"         
            "faultBit"          
            "warningBit"    
            "faultValue"     
            "warningValue"    
            "constantPowerOK"   
            "epvtoday"          
            "pvenergytoday"    
            "epvtotal"          
            "eacCharToday"     
            "eacCharTotal"      
            "ebatDischarToday"  
            "ebatDischarTotal"  
            "eacDischarToday"  
            "eacDischarTotal"   
            "ACCharCurr"        
            "ACDischarWatt"    
            "ACDischarVA"       
            "BatDischarWatt"   
            "BatDischarVA"      
            "BatWatt"           
       
I tried to match this as much as possible with the growatt fields (website / documentation: https://docplayer.net/199800433-Growatt-offgrid-spf5000-modbus-rs485-rtu-protocol.html). 

But the mapping might be not complete (I see by example I mis output DCV) but that can be correct if needed. Same values are copied in 2 fields (because of compatibility).  

I am not sure which values you need (are interesting). I should go for (first try): pvpowerout (= same as op_watt), epvtoday (=pvenergytoday), epvtotal, pvgridvoltage (=outputvolt), line_freq).  invtemp 

But i do understand you also want to display battery information. I only do not know which fields are of interest (I do no have a spf/battery).

**Reply by Dandiiik** [reply]
Thanks for the quick response and help. Yes I have 3 inverters the same and 3x Shine Wifi S
I see the data, but when it shows it in graphs, the values ​​are different. Incorrect values ​​appear to be assigned. Or they look and they don't show up.
I want to try to run 3x Grott with different port. Or what else could I try?

**Comment by johanmeijer** [comment]
Maybe 3x grott might be a solution. I know grott works with one shinelink and multiple inverters connected with  rf. 3 Separate dataloggers I have not tested yet (comming soon I have a second one bought) buf I have seen unexpected behaviour (see also issue #110).

In future I want to improve multi threading for Grott.
